src/client.js

```diff
@@ -572,6 +572,27 @@ MatrixClient.prototype.rehydrateDevice = async function() {
     }
 };
 
+/**
+ * Get the current dehydrated device, if any
+ * @return {Promise} A promise of an object containing the dehydrated device
+ */
+MatrixClient.prototype.getDehydratedDevice = async function() {
+    try {
+        return await this._http.authedRequest(
```

Comment:
This same code block exists in `MatrixClient.prototype.rehydrateDevice`, could you update it to call the new function you've added?